android/build.gradle

```diff
@@ -19,7 +19,57 @@ buildscript {
 
 apply plugin: "com.android.library"
 
+def resolveReactNativeDirectory() {
+    // monorepo workaround
+    // react-native can be hoisted or in project's own node_modules
+    def reactNativeFromProjectNodeModules = file("${rootProject.projectDir}/../node_modules/react-native")
+    if (reactNativeFromProjectNodeModules.exists()) {
+        return reactNativeFromProjectNodeModules
+    }
+
+    def reactNativeFromNodeModulesWithLibrary = file("${projectDir}/../../react-native")
+    if (reactNativeFromNodeModulesWithLibrary.exists()) {
+        return reactNativeFromNodeModulesWithLibrary
+    }
+
+    throw new Exception(
+            "[react-native-clipboard] Unable to resolve react-native location in " +
+                    "node_modules. You should add project extension property (in app/build.gradle) " +
+                    "`REACT_NATIVE_NODE_MODULES_DIR` with path to react-native."
+    )
+}
```

Comment:
I am working with a monorepo using [NPM workspaces](https://docs.npmjs.com/cli/v9/using-npm/workspaces?v=true) and I can confirm that this implementation of `resolveReactNativeDirectory()` does not enable monorepo support.

## Problem

The values for both `reactNativeFromProjectNodeModules` and `reactNativeFromNodeModulesWithLibrary` are the same!

In a monorepo setup there is likely to be a `node_modules` local to the package's folder, **and** a `node_modules` at the root of the monorepo. I suspect that the intention of having both a `reactNativeFromProjectNodeModules` and `reactNativeFromNodeModulesWithLibrary` was for one to point to each of the two possible locations where the `react-native` library might be located.

## Setup

My workspaces is configured as such in the root `package.json`:

```json
     "workspaces": [
       "apps/*"
     ]
```

I have a folder structure like this:

```
.
├── apps
│   └── rn-app
│       ├── android
│       ├── ios
│       ├── node_modules
│       │   └── react-native-permissions
│       ├── src
│       ├── index.js
│       ├── package.json
│       ├── react-native.config.js
│       └── tsconfig.json
├── node_modules
│   ├── react-native
│   │   └── ...
│   └── ...
├── README.md
├── package-lock.json
└── package.json
```

And, in the file, `apps/rn-app/package.json` I have a reference to `react-native-permissions@3.8.4` under the `dependencies`. 

## Cause

Due to how NPM is resolving my dependencies my reference to `react-native-permissions` has not been hoisted to the `node_modules` at the root of my project (although `react-native` has been hoisted). Consequently, the exception message is triggering. Alas, I have no idea what the instruction of the exception message is asking me to do.

Reply:
@trippingtarballs Hello!
I also have a monorepo and everything works for me. take a look at my implementation.
https://github.com/serserm/react-native-turbo-sensors

modified files:
[react-native.config.js](https://github.com/serserm/react-native-turbo-sensors/blob/main/example/react-native.config.js)
[metro.config.js](https://github.com/serserm/react-native-turbo-sensors/blob/main/example/metro.config.js)
[babel.config.js](https://github.com/serserm/react-native-turbo-sensors/blob/main/example/babel.config.js)
[settings.gradle](https://github.com/serserm/react-native-turbo-sensors/blob/main/example/android/settings.gradle)
[settings.gradle](https://github.com/serserm/react-native-turbo-sensors/blob/main/packages/react-native-turbo-sensors/android/settings.gradle)
[tsconfig.json](https://github.com/serserm/react-native-turbo-sensors/blob/main/packages/react-native-turbo-sensors/tsconfig.json)